pkg/azclient/auth_conf.go

```diff
@@ -46,6 +48,13 @@ type AzureAuthConfig struct {
 	// Auxiliary token provider for accessing resources from network tenant
 	// Require MSI to be enabled and have permission to access the KeyVault
 	AuxiliaryTokenProvider *AzureAuthAuxiliaryTokenProvider `json:"auxiliaryTokenProvider,omitempty" yaml:"auxiliaryTokenProvider,omitempty"`
+	// The path where a JSON file exists containing the JSON format of a UserAssignedIdentityCredentials struct
+	// See the msi-dataplane for more details on UserAssignedIdentityCredentials - https://github.com/Azure/msi-dataplane/blob/63fb37d3a1aaac130120624674df795d2e088083/pkg/dataplane/internal/generated_client.go#L156C6-L156C37
+	AADMSIDataPlaneIdentityPath string `json:"aadMSIDataPlaneIdentityPath,omitempty" yaml:"aadMSIDataPlaneIdentityPath,omitempty"`
+	// Used with AADMSIDataPlaneIdentityPath to specify the Cloud type.
+	// Can only be one of the following values: AzurePublicCloud, AzureUSGovernmentCloud, or AzureChinaCloud
+	// If a value is not specified, defaults to AzurePublicCloud
+	MSIDataPlaneIdentityCloudType dataplane.AzureCloud `json:"msIDataPlaneIdentityCloudType,omitempty" yaml:"msIDataPlaneIdentityCloudType,omitempty"`
```

Comment:
could we infer it from config `Cloud`? values should be same here

Reply:
Sure